host/retry_policy_workflow_test.go

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package host
+
+import (
+	"bytes"
+	"strconv"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/log/tag"
+	"github.com/uber/cadence/common/types"
+)
+
+func (s *IntegrationSuite) TestWorkflowRetryPolicyTimeout() {
+	id := "integration-workflow-retry-policy-timeout-test"
+	wt := "integration-workflow-retry-policy-timeout-test-type"
+	tl := "integration-workflow-retry-policy-timeout-test-tasklist"
+	identity := "worker1"
+
+	workflowType := &types.WorkflowType{}
+	workflowType.Name = wt
+
+	taskList := &types.TaskList{}
+	taskList.Name = tl
+
+	retryPolicy := &types.RetryPolicy{
+		InitialIntervalInSeconds:    1,
+		BackoffCoefficient:          2.0,
+		MaximumIntervalInSeconds:    5,
+		ExpirationIntervalInSeconds: 5,
+		MaximumAttempts:             0,
+		NonRetriableErrorReasons:    []string{"bad-error"},
+	}
+	request := &types.StartWorkflowExecutionRequest{
+		RequestID:                           uuid.New(),
+		Domain:                              s.domainName,
+		WorkflowID:                          id,
+		WorkflowType:                        workflowType,
+		TaskList:                            taskList,
+		Input:                               nil,
+		ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(10),
+		TaskStartToCloseTimeoutSeconds:      common.Int32Ptr(10),
+		Identity:                            identity,
+		RetryPolicy:                         retryPolicy,
+	}
+	now := time.Now()
+	we, err0 := s.engine.StartWorkflowExecution(createContext(), request)
+	s.Nil(err0)
+
+	s.Logger.Info("StartWorkflowExecution", tag.WorkflowRunID(we.RunID))
+	time.Sleep(11 * time.Second) // wait 1 second for timeout
```

Comment:
Can we use a smaller workflow timeout like 3s? Wait 10s here seems a bit long for a test, and all three tests need 35s to complete.